test/new-e2e/tests/discovery/linux_test.go

```diff
@@ -0,0 +1,131 @@
+var systemProbeConfigStr string
+
+//go:embed testdata/config/check_config.yaml
+var checkConfigStr string
+
+type linuxTestSuite struct {
+	e2e.BaseSuite[environments.Host]
+}
+
+func TestLinuxTestSuite(t *testing.T) {
+	agentParams := []func(*agentparams.Params) error{
+		agentparams.WithAgentConfig(agentConfigStr),
+		agentparams.WithSystemProbeConfig(systemProbeConfigStr),
+		agentparams.WithFile("/etc/datadog-agent/conf.d/service_discovery.d/conf.yaml", checkConfigStr, true),
+	}
+	options := []e2e.SuiteOption{
+		e2e.WithProvisioner(awshost.Provisioner(awshost.WithAgentOptions(agentParams...))),
+	}
+	e2e.Run(t, &linuxTestSuite{}, options...)
+}
+
+func (s *linuxTestSuite) SetupSuite() {
+	s.BaseSuite.SetupSuite()
+
+	s.provisionServer()
+}
+
+func (s *linuxTestSuite) TestServiceDiscoveryCheck() {
+	t := s.T()
+	s.startServices()
+	defer s.stopServices()
+
+	client := s.Env().FakeIntake.Client()
+	err := client.FlushServerAndResetAggregators()
+	require.NoError(t, err)
+
+	assert.EventuallyWithT(t, func(t *assert.CollectT) {
+		assertRunningCheck(t, s.Env().RemoteHost, "service_discovery")
+	}, 2*time.Minute, 10*time.Second)
+
+	// This is very useful for debugging, but we probably don't want to decode
+	// and assert based on this in this E2E test since this is an internal
+	// interface between the agent and system-probe.
+	services := s.Env().RemoteHost.MustExecute("sudo curl -s --unix /opt/datadog-agent/run/sysprobe.sock http://unix/discovery/services")
+	t.Log("system-probe services", services)
+
+	assert.EventuallyWithT(t, func(c *assert.CollectT) {
+		payloads, err := client.GetServiceDiscoveries()
+		require.NoError(t, err)
+
+		found := false
+		for _, p := range payloads {
+			name := p.Payload.ServiceName
+			t.Log("RequestType", p.RequestType, "ServiceName", name)
+
+			if p.RequestType == "start-service" && name == "python" {
+				found = true
+				break
+			}
+		}
+
+		assert.True(c, found, "service not found")
+	}, 3*time.Minute, 10*time.Second)
+}
+
+type checkStatus struct {
+	CheckID           string `json:"CheckID"`
+	CheckName         string `json:"CheckName"`
+	CheckConfigSource string `json:"CheckConfigSource"`
+	ExecutionTimes    []int  `json:"ExecutionTimes"`
+}
+
+type runnerStats struct {
+	Checks map[string]checkStatus `json:"Checks"`
+}
+
+type collectorStatus struct {
+	RunnerStats runnerStats `json:"runnerStats"`
+}
+
+// assertRunningCheck asserts that the given process agent check is running
+func assertRunningCheck(t *assert.CollectT, remoteHost *components.RemoteHost, check string) {
+	statusOutput := remoteHost.MustExecute("sudo datadog-agent status collector --json")
+
+	var status collectorStatus
+	err := json.Unmarshal([]byte(statusOutput), &status)
+	require.NoError(t, err, "failed to unmarshal agent status")
+
+	assert.Contains(t, status.RunnerStats.Checks, check)
+}
+
+func (s *linuxTestSuite) provisionServer() {
+	err := s.Env().RemoteHost.CopyFolder("testdata/provision", "/home/ubuntu/e2e-test")
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
You can copy this at pulumi time, but you'd need a custom run function instead of the out of the box provisioner. You can keep this is as, we might consider adding an helper for the vm options.

Reply:
Here is an example of a file system operation at pulumi time https://github.com/DataDog/datadog-agent/blob/main/test/new-e2e/examples/customenv_with_filemanager_test.go#L41